app/Services/UpdateOfferService.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Requests\OfferRequest;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Models\Offer;
+use App\Repositories\OfferCategoryRepository;
+use App\Repositories\OfferRepository;
+use Exception;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\Response;
+
+class UpdateOfferService
+{
+    protected $offerRepository;
+    protected $categoryRepository;
+
+    public function __construct(OfferRepository $offerRepository, OfferCategoryRepository $categoryRepository)
+    {
+        $this->offerRepository = $offerRepository;
+        $this->categoryRepository = $categoryRepository;
+    }
+
+    public function execute(OfferRequest $request, Offer $offer)
+    {
+        try {
+            $this->categoryRepository->findOrFail($request->category_id);
+
+            $offer->category_id = $request->category_id;
+            $offer->start_at = $request->start_at;
+            $offer->end_at = $request->end_at;
+            $offer->price_per_hour = $request->price_per_hour;
+            $offer->address = $request->address;
+            $offer->preferred_qualifications = $request->preferred_qualifications;
+            $offer->save();
```

Comment:
@RanaEmad Why not using the built-in update method of the model instead of updating each value manually.